pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -2644,37 +2645,41 @@ func (r *ClickHouseReader) GetDisks(ctx context.Context) (*[]model.DiskItem, *mo
 			fmt.Errorf("error while getting disks. Err=%v", err)}
 	}
 
-
 	zap.S().Infof("Got response: %+v\n", diskItems)
 
 	return &diskItems, nil
 }
 
 func (r *ClickHouseReader) GetTTL(ctx context.Context, ttlParams *model.GetTTLParams) (*model.GetTTLResponseItem, *model.ApiError) {
 
-	parseTTL := func(queryResp string) int {
-		values := strings.Split(queryResp, " ")
-		N := len(values)
-		ttlIdx := -1
+	parseTTL := func(queryResp string) (int, int) {
+
+		zap.S().Debugf("Parsing TTL from: %s", queryResp)
+		deleteTTLExp := regexp.MustCompile(`TTL toDateTime\(timestamp\) \+ toIntervalSecond\(([0-9]*)\)`)
```

Comment:
Is there an alternative way of achieving this? This could be `toIntervalDay` or `toIntervalMonth` which would not work.

Reply:
There is no alternate way other than parsing the create table request. Clickhouse doesn't; provide a way to query for TTL, sadly.

Why would it be `toIntervalDay` or `toIntervalMonth`? It's always going to be `toIntervalSecond` because we are creating the TTL by using `toIntervalSecond` in the setTTL request.

Reply:
There is another PR where we are trying to set the default ttl in days and months. I think changing them to seconds should be fine if we agreed on to set the ttl only in seconds precision.

Reply:
Yeah. Extracting information from create table string is definitely not a clean way. But clickhouse doesn't have any other way yet.